### PR TITLE
Bubble_sort algorithm

### PR DESCRIPTION
**Pull request by brytebee**
## Bubble  Sort Project 
- In this project, I create a sorting algorithm using the core Ruby programming language. The algorithm receives an array of numbers and re-arranges the array in ascending order.

> Do leave any necessary modification suggestions.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @brytebee, 👋🏽 
 **Thank you for the good job you got going on. The  method runs just fine and almost pass all test cases.**

## REQUIRED CHANGES

- [x] You have 1 rubocop offense detected locally.
- [x] Kindly add a brief description for this PR.
- [x] Kindly add a second method `bubble_sort_by` as required [in ordin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#assignment-1). See [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png). So you need two methods.

**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.

# Reach me on slack for a chitchat 🆘  **Roy Mukuye**

**Comment by brytebee** [comment]
I have effected the required changes. The Bubble_sort_by method is in a separate file.

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Bright Atsighi** 
Nice job with the improvements for this project :muscle:  :muscle:
Your algorithms passes most of the tests :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [ ] One of your methods is not working properly. Check out the expected behavior:
`bubble_sort_by(%w[functioning? is function this]) { |left, right| left.length - right.length } #expected this to be: ["is", "this", "function", "functioning?"]`
- [ ] Add a screenshot in your README file
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  swap = 0
+  arr_length.times do
+    (0..(arr_length - 1)).each do |i|
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap += 1
+      end
+    end
+  end
+  puts swap
```
Comment (bini-i) [inline bubble_sort.rb:16]: kindly remove debugging instructions as it is not related to the output of the function

File: bubble_sort_by.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  swap = 0
+  arr_length.times do
+    (0..(arr_length - 1)).each do |i|
+      if yield(arr[i], arr[i + 1]).negative?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap += 1
+      end
+    end
+  end
+  puts swap
```
Comment (bini-i) [inline bubble_sort_by.rb:17]: Kindly remove debugging instructions as it is not related to the output of the function.

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle: 
### Hello  @brytebee :wave: 

### Thanks for working on the changes suggested. You have done a great job and your project passes most of the requirements, but you still can improve your project. 

- [ ] bubble_sory_by sorts an array in reverse order, Please make sure it sorts in ascending order.
THe following is the result I got as I run your code with examples provided [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421588/)
Screenshot from 2021-02-23 07-19-19

- [ ] and remove debugging instruction in committed code. Please refer to Inline comments below :point_down: .
- [ ] Kindly add how to use the repo section in your README to illustrate how to clone your code and run it on a local machine
#### If you have any question kindly reach me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (TSE)
> GitHub: [@bini-i](https://github.com/bini-i)
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Review by mke2111 — APPROVED** [review]
# **🎉 STATUS: APPROVED 🎉** 

## Hey @brytebee, 👋🏽 
 **Congratulations 🥳 🥂 . The methods run as expected.**

**You may now merge to the main branch and submit a project completion form. 👍🏽** 

## Cheers!! 🍹 

## Reach me on slack for a chitchat 🆘  **Roy Mukuye**

**Comment by brytebee** [comment]
Thank you.

**Pull request merged**